src/main/java/com/twilio/taskrouter/domain/model/WorkspaceFacade.java

```diff
@@ -51,34 +54,45 @@ public static WorkspaceFacade create(TwilioTaskRouterClient twilioTaskRouterClie
     }
   }
 
+  public static Optional<WorkspaceFacade> findBySid(String workspaceSid, TwilioTaskRouterClient
+    twilioTaskRouterClient) {
+    return twilioTaskRouterClient.getWorkspaces().getPageData().stream()
+      .filter(workspace -> workspace.getSid().equals(workspaceSid)).findFirst()
+      .map(workspace -> new WorkspaceFacade(twilioTaskRouterClient, workspace));
+  }
+
   public String getFriendlyName() {
     return workspace.getFriendlyName();
   }
 
-  public Worker addWorker(Map<String, String> parameters) {
+  public String getSid() {
+    return workspace.getSid();
+  }
+
+  public Worker addWorker(Map<String, String> newWorkerParams) {
     try {
 
-      return workspace.createWorker(parameters);
+      return workspace.createWorker(newWorkerParams);
```

Comment:
Same comment about naming. IMO `new` is adding noise. Please take care of this one as well.